src/gpu/multiexp.rs

```diff
@@ -304,28 +304,28 @@ where
 
         let mut acc = <G as CurveAffine>::Projective::zero();
 
-        let results = crate::multicore::RAYON_THREAD_POOL.install(|| {
-            if n > 0 {
-                bases
+        let results = if n > 0 {
```

Comment:
It's good if it fixes the deadlock. Though it shouldn't be a permanent fix. The things are still running in a rayon thread pool, the difference is that it is not the `crate::multicore::RAYON_THREAD_POOL`, but a global one. Ideally the `RAYON_THREAD_POOL` shouldn't deadlock (we should check why it does and also document the thread pools better, to be clearer which one is used for which things.